File: src/org/genyris/core/Biscuit.java
package org.genyris.core;

import org.genyris.exception.GenyrisException;
import org.genyris.interp.Environment;

public class Biscuit extends Atom  implements Comparable{


    private Exp expression;

    public Biscuit(Exp expression) {
       this.expression = expression;
    }

    @Override
    public String toString() {
        return "<" + this.getClass().getName() + " " + expression.toString() + ">";
    }

    @Override
    public void acceptVisitor(Visitor guest) throws GenyrisException {
        guest.visitBiscuit(this);  
    }

    @Override
    public Exp eval(Environment env) throws GenyrisException {
        return this;
    }

    @Override
    public Symbol getBuiltinClassSymbol(Internable table) {
        return table.BISCUIT();
    }

    public Exp getExpression() {
        return expression;
    }

        @Override
    public int compareTo(Object o) {
        return this == o ? 0 : 1;
    }


}

File: src/org/genyris/core/Biscuit.kt
package org.genyris.core

import org.genyris.exception.GenyrisException
import org.genyris.interp.Environment

class Biscuit(val expression: Exp) : Atom(), Comparable<Any?> {
    override fun toString(): String {
        return "<" + this.getClass().getName() + " " + expression.toString() + ">"
    }

    @kotlin.Throws(GenyrisException::class)
    override fun acceptVisitor(guest: Visitor) {
        guest.visitBiscuit(this)
    }

    @kotlin.Throws(GenyrisException::class)
    override fun eval(env: Environment?): Exp {
        return this
    }

    override fun getBuiltinClassSymbol(table: Internable): Symbol? {
        return table.BISCUIT()
    }

    override fun compareTo(o: Any?): Int {
        return if (this === o) 0 else 1
    }
}
